api/src/main/java/org/eclipse/microprofile/concurrent/ManagedExecutorConfig.java

```diff
@@ -19,27 +19,42 @@
 package org.eclipse.microprofile.concurrent;
 
 import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE;
 import static java.lang.annotation.RetentionPolicy.RUNTIME;
 
 import java.lang.annotation.Retention;
 import java.lang.annotation.Target;
 
+import javax.inject.Qualifier;
+
 /**
- * <p>Configures an injected <code>ManagedExecutor</code> instance.</p>
+ * <p>Configures an injected {@link ManagedExecutor} instance, which is
+ * {@link javax.enterprise.context.ApplicationScoped application scoped}.
+ * Each uniquely qualified configuration within an application corresponds
+ * to a different {@link ManagedExecutor} instance.
+ * When beans within an application specify identical attributes for the
+ * <code>ManagedExecutorConfig</code> qualifier, the same instance is injected.</p>
  *
  * <p>Example usage:</p>
  * <pre><code> &commat;Inject &commat;ManagedExecutorConfig(propagated=ThreadContext.CDI, maxAsync=5)
  * ManagedExecutor executor;
  * ...
  * </code></pre>
  *
- * <p>A <code>ManagedExecutor</code> must fail to inject, raising
+ * <p>A {@link ManagedExecutor}'s life cycle is tied to that of the application.
+ * When the application stops, the container automatically shuts down the
+ * managed executor and cancels its remaining actions/tasks.</p>
+ *
+ * <p>A {@link ManagedExecutor} must fail to inject, raising
  * {@link javax.enterprise.inject.spi.DeploymentException DeploymentException}
  * on application startup, if more than one provider provides the same thread context
  * {@link org.eclipse.microprofile.concurrent.spi.ThreadContextProvider#getThreadContextType type}.
  */
+@Qualifier
 @Retention(RUNTIME)
-@Target(FIELD)
+@Target({ FIELD, METHOD, PARAMETER, TYPE })
```

Comment:
I couldn't think of a reason for wanting TYPE either, but added it after reviewing documentation such as 
https://javaee.github.io/tutorial/cdi-basic006.html
and
https://docs.oracle.com/javaee/6/tutorial/doc/gjbck.html
which states that, "A qualifier type is a Java annotation defined as @Target({METHOD, FIELD, PARAMETER, TYPE}) and @Retention(RUNTIME)."
This seemed to suggest TYPE is required for qualifiers.
However, it appears [other documentation](https://docs.jboss.org/cdi/spec/1.2/cdi-spec.html) uses the term "typically", meaning it isn't a requirement: 
"A qualifier type is a Java annotation defined as @Retention(RUNTIME). Typically a qualifier type is defined as @Target({METHOD, FIELD, PARAMETER, TYPE})."

Let's have Matej, who is an expert in CDI, and also involved in this spec review this.

Reply:
You are _not required_ to have `TYPE` there.
CDI specification uses the word ["typically"](http://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#defining_qualifier_types) because most of the time you want to be able to place such qualifier on a bean backed by a Java class.

It is completely up to you what do you actually restrict it to. Then again, there is no real drawback to keeping `TYPE` there and you may, theoretically, have a class using it (a implementation of `ManagedExecutor` with this qualifier on it).